src/lib/connector.ts

```diff
@@ -37,9 +38,9 @@ export interface ConnectOptions {
  * @param options Options to control client connection.
  * @returns Promise that resolves to a Connection object.
  */
-export async function connect(options: ConnectOptions): Promise<Connection> {
+export function connect(options: ConnectOptions): Connection {
```

Comment:
This is really nice! While slightly breaking (in terms of docs/usage), I think it is going to improve the DX over the long run. Nice find/fix.